python_modules/dagster/dagster/_core/definitions/sensor_definition.py

```diff
@@ -231,13 +288,75 @@ def log_key(self) -> Optional[List[str]]:
     RunRequest,
     PipelineRunReaction,
 ]
-RawSensorEvaluationFunction = Union[
-    Callable[[], RawSensorEvaluationFunctionReturn],
-    Callable[[SensorEvaluationContext], RawSensorEvaluationFunctionReturn],
-]
-SensorEvaluationFunction = Callable[
-    [SensorEvaluationContext], Iterator[Union[SkipReason, RunRequest]]
-]
+RawSensorEvaluationFunction: TypeAlias = Callable[..., RawSensorEvaluationFunctionReturn]
+
+SensorEvaluationFunction: TypeAlias = Callable[..., Iterator[Union[SkipReason, RunRequest]]]
+
+
+def context_param_name_if_present(fn: Callable) -> Optional[str]:
+    """
+    Determines the sensor's context parameter name by excluding all resource parameters.
+    """
+    resource_params = {param.name for param in get_resource_args(fn)}
+
+    return next(
+        (param.name for param in get_function_params(fn) if param.name not in resource_params), None
+    )
+
+
+def _validate_and_get_resource_dict(
+    context: SensorEvaluationContext, sensor_name: str, required_resource_keys: Set[str]
+) -> Dict[str, Any]:
+    """
+    Validates that the context has all the required resources and returns a dictionary of
+    resource key to resource object.
+    """
+    for k in required_resource_keys:
+        if not hasattr(context.resources, k):
+            raise DagsterInvalidDefinitionError(
+                f"Resource with key '{k}' required by sensor '{sensor_name}' was not provided."
+            )
+
+    return {k: getattr(context.resources, k) for k in required_resource_keys}
+
+
+def get_or_create_sensor_context(
+    fn: Callable, *args: Any, **kwargs: Any
+) -> SensorEvaluationContext:
+    """
+    Based on the passed resource function and the arguments passed to it, returns the
+    user-passed SensorEvaluationContext or creates one if it is not passed.
+
+    Raises an exception if the user passes more than one argument or if the user-provided
+    function requires a context parameter but none is passed.
+    """
+    context_param_name = context_param_name_if_present(fn)
```

Comment:
btw with the rename suggestion i was referring to the variable name, not the function name. it ends up making code further down make more sense:

`context_param_name = context_param_name or list(kwargs.keys())[0]`

becomes

`context_param_name = context_param_name_if_present or list(kwargs.keys())[0]`

as an example